### DISCUSSION TITLE
Excessive gap between icons on desktop

### DISCUSSION THREAD

**Discussion by ffs-akl** [lxqt/lxqt#2611] [Q&A] [answered]
Under the Locale zh_CN.UTF8,   Excessive gap between icons on desktop.   if chang to en_US.UTF8,  The gap between the icons on desktop will be much more normalized.
![swappy-20240916-194251](https://github.com/user-attachments/assets/46733b32-0b3c-43e7-8309-c2dff545650b)

**Comment by stefonarch** [comment]
No idea, here I see actually the opposite:

![niri_16 14-34-31](https://github.com/user-attachments/assets/27ef695b-0708-4d1d-b318-f0ac8c3464e5) ![niri_16 14-35-28](https://github.com/user-attachments/assets/9735cba7-e609-4ed6-8d76-aa3ca05568e5)

**Reply by ffs-akl** [reply]
Lady Luck must like you!  this issue and  issue of portal doesn't exist with you.

**Reply by ffs-akl** [reply]
Thank you for your reply！

**Reply by tsujan** [reply]
> Lady Luck must like you

I agree about the portal issue ;) You and I see it; @stefonarch doesn't. I use this workaround in a startup script under kwin_wayland (labwc is OK):

```sh
  if ! pidof "xdg-desktop-portal-lxqt" 1 > /dev/null; then
    systemctl --user restart xdg-desktop-portal.service
  fi
```

But this one is NOT an issue.

**Reply by ffs-akl** [reply]
Yes，I agree.

**Comment by tsujan** [comment, marked as answer]
https://github.com/lxqt/libfm-qt/blob/eae367aae55d595696edd1f615483028569756a2/src/folderview.cpp#L1215-L1218:
```c++
        // However, this average only applies to English. For some Asian characters, such as Chinese chars,
        // each char actually takes doubled space. To be safe, we use 13 chars per line x average char width
        // to get a nearly optimal width for the text label. As most of the filenames have less than 40 chars
        // 13 chars x 3 lines should be enough to show the full filenames for most files.
```

**Reply by ffs-akl** [reply]
So, there is no way to chang the gap between icons on desktop by setting it manually.

**Reply by tsujan** [reply]
There's no *real* gap.

Three lines of text are always reserved, and if the available screen height doesn't allow `n+1` items in each column, `n` items are spread vertically such that no extra gap appears at the bottom. Therefore, you may see gaps depending on font size, icon size, minimum item margins, and your available screen height (the screen height minus the heights of horizontal panels).

**Reply by ffs-akl** [reply]
yes,  It is.  thank you very much!
